### PR TITLE
[ut] add ut for pkg/utils/namespace

### PR DESCRIPTION
**Pull request by zhuwint**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

add ut for pkg/utils/namespace

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

pkg/utils/namespace.go coverage 100%

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by RainbowMango** [comment]
cc @Poor12 Please help to take a look.

**Comment by RainbowMango** [comment]
@zhuwint thanks for doing this. 
Please add your email to [GitHub Email](https://github.com/settings/emails), otherwise, GitHub can't recognize you, and thus can't count your contribution.

**Comment by zhuwint** [comment]
> 

ok

**Comment by zhuwint** [comment]
> @zhuwint thanks for doing this. Please add your email to [GitHub Email](https://github.com/settings/emails), otherwise, GitHub can't recognize you, and thus can't count your contribution.

ok

File: pkg/util/namespace_test.go [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		want    bool
+		wantErr bool
+	}{
+		{
+			name: "query error",
```
Comment (Poor12) [inline pkg/util/namespace_test.go:30]: query namespace error

File: pkg/util/namespace_test.go [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+				namespace: "default",
+				reactor: func(action coretesting.Action) (handled bool, ret runtime.Object, err error) {
+					return true, &corev1.Namespace{}, errors.New("fail get namespace")
+				},
+			},
+			want:    false,
+			wantErr: true,
+		},
+		{
+			name: "not exist",
```
Comment (Poor12) [inline pkg/util/namespace_test.go:42]: namespace not exist

File: pkg/util/namespace_test.go [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+			args: args{
+				client:    fake.NewSimpleClientset(&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "default-1"}}),
+				namespace: "default",
+				reactor:   nil,
+			},
+			want:    false,
+			wantErr: false,
+		},
+		{
+			name: "exist",
```
Comment (Poor12) [inline pkg/util/namespace_test.go:52]: namespace already exists

**Review by Poor12 — COMMENTED** [review]
Please check all name of testcases, to make more readable.

File: pkg/util/namespace_test.go [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+		want    bool
+		wantErr bool
+	}{
+		{
+			name: "query error",
+			args: args{
+				client:    fake.NewSimpleClientset(&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "default"}}),
+				namespace: "default",
+				reactor: func(action coretesting.Action) (handled bool, ret runtime.Object, err error) {
+					return true, &corev1.Namespace{}, errors.New("fail get namespace")
```
[the remark this answers is not in the document]
Comment (zhuwint) [reply pkg/util/namespace_test.go]: ok, it's modified

File: pkg/util/namespace_test.go (hunk 1 above)
Comment (zhuwint) [reply pkg/util/namespace_test.go]: ok, it's modified

File: pkg/util/namespace_test.go (hunk 2 above)
Comment (zhuwint) [reply pkg/util/namespace_test.go]: ok, it's modified

File: pkg/util/namespace_test.go (hunk 3 above)
Comment (zhuwint) [reply pkg/util/namespace_test.go]: ok, it's modified

**Review by XiShanYongYe-Chang — COMMENTED** [review]
@zhuwint, good job!

/lgtm

File: pkg/util/namespace_test.go [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+	coretesting "k8s.io/client-go/testing"
+)
+
+func TestIsNamespaceExist(t *testing.T) {
+	type args struct {
+		client    kubernetes.Interface
```
Comment (RainbowMango) [inline pkg/util/namespace_test.go:18]:
```suggestion
		client    *fake.Clientset
```
If use `*fake.Clientset` here, then you don't need to convert with `tt.args.client.(*fake.Clientset)`.

File: pkg/util/namespace_test.go [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.args.reactor != nil {
+				tt.args.client.(*fake.Clientset).PrependReactor("get", "namespaces", tt.args.reactor)
+			}
+
+			got, err := IsNamespaceExist(tt.args.client, tt.args.namespace)
+			if (err != nil) != tt.wantErr {
```
Comment (RainbowMango) [inline pkg/util/namespace_test.go:69]:
```suggestion
			if (err == nil && tt.wantErr == true) || (err != nil && tt.wantErr == false) {
```
Suggestion for readability.

**Review by RainbowMango — COMMENTED** [review]
Generally looks good to me. 
Just some nits, otherwise LGTM.

File: pkg/util/namespace_test.go (hunk 5 above)
Comment (zhuwint) [reply pkg/util/namespace_test.go]: Ok, it's modified, thanks for pointing it out

File: pkg/util/namespace_test.go (hunk 6 above)
Comment (zhuwint) [reply pkg/util/namespace_test.go]: Ok, it's modified, thanks for pointing it out

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**